contracts/utils/Strings.sol

```diff
@@ -63,17 +64,15 @@ library Strings {
      * @dev Converts a `uint256` to its ASCII `string` hexadecimal representation with fixed length.
      */
     function toHexString(uint256 value, uint256 length) internal pure returns (string memory) {
-        uint256 localValue = value;
+        if (length < Math.log256(value) + 1) {
```

Comment:
computing the log is quite expensive. Checking at the end is less expensive (when everything works fine, which is the case we should optimize for). Any reason to not keep it?